### PR TITLE
[DPS-499] Generate data products and source apps

### PR DESCRIPTION
**Pull request by jamessnowplow**
by the power of any...

### CODE REVIEW

File: cmd/dp/generate.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	"github.com/flytam/filenamify"
+	"github.com/google/uuid"
+	snplog "github.com/snowplow-product/snowplow-cli/internal/logging"
+	"github.com/snowplow-product/snowplow-cli/internal/util"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+)
+
+func mkSafeFilename(name string) (string, error) {
+	appn, err := filenamify.FilenamifyV2(name)
```
Comment (gleb-lobov) [inline cmd/dp/generate.go:29]: I think I did a decent job with `ResourceNameToFileName`, do we need this one?

File: go.mod [hunk 2]
```diff
@@ -4,13 +4,15 @@ go 1.22.6
...
 	github.com/charmbracelet/log v0.4.0
+	github.com/flytam/filenamify v1.2.0
 	github.com/go-playground/validator/v10 v10.22.1
 	github.com/go-viper/mapstructure/v2 v2.1.0
 	github.com/google/uuid v1.6.0
 	github.com/r3labs/diff/v3 v3.0.1
 	github.com/santhosh-tekuri/jsonschema/v5 v5.3.1
 	github.com/spf13/cobra v1.8.1
 	github.com/spf13/pflag v1.0.5
+	golang.org/x/net v0.23.0
```
Comment (gleb-lobov) [inline go.mod:15]: Not sure if I understand why do we need it

File: cmd/dp/generate.go (hunk 1 above)
Comment (gleb-lobov) [reply cmd/dp/generate.go:29]: Also it will make it consistent with how `download` names files

File: cmd/dp/generate.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+			if err != nil {
+				snplog.LogFatal(err)
+			}
+			slog.Info("generate", "msg", "wrote", "kind", "data product", "file", filename)
+		}
+
+	},
+}
+
+func write(tpl any, fname string, format string) error {
```
Comment (gleb-lobov) [inline cmd/dp/generate.go:98]: Maybe we can use `util.WriteSerializableToFile` here?

File: cmd/dp/generate.go (hunk 1 above)
Comment (jamessnowplow) [reply cmd/dp/generate.go:29]: looking at the two I think the only bits we dont cover are max length (dubious value) and the windows reserved ones https://github.com/flytam/filenamify/blob/master/filenamify.go#L26 we could just add the windows ones in ours?

**Review by gleb-lobov — APPROVED** [review]

**Pull request merged**